src/ra/groupby.h

```diff
@@ -0,0 +1,110 @@
+#ifndef RA_GroupBy_H_
+#define RA_GroupBy_H_
+
+#include <type_traits>
+#include <utility>
+#include <map>
+
+#include <iostream>
+
+#include "range/v3/all.hpp"
+
+namespace fluent {
+namespace ra {
+
+struct hash_fn {
```

Comment:
This shouldn't be called `hash_fn` if there's no hashing going on. Could this whole thing also be replaced with something like `rng | view::transform([](const auto& t) { return a pair of key and value}) | to_<std::map>`?